code/datums/components/pellet_cloud.dm

```diff
@@ -0,0 +1,249 @@
+	var/targloc = get_turf(target)
+	if(!zone_override)
+		zone_override = shooter.zone_selected
+
+	for(var/i in 1 to num_pellets)
+		A.ready_proj(target, user, SUPPRESSED_VERY, zone_override, fired_from)
+		if(distro)
+			if(randomspread)
+				spread = round((rand() - 0.5) * distro)
+			else //Smart spread
+				spread = round((i / num_pellets - 0.5) * distro)
+
+		RegisterSignal(A.BB, COMSIG_PROJECTILE_SELF_ON_HIT, .proc/pellet_hit)
+		RegisterSignal(A.BB, list(COMSIG_PROJECTILE_RANGE_OUT, COMSIG_PARENT_QDELETING), .proc/pellet_range)
+		pellets += A.BB
+		if(!A.throw_proj(target, targloc, shooter, params, spread))
+			return 0
+		if(i != num_pellets)
+			A.newshot()
+
+/**
+  * create_blast_pellets() is for when we have a central point we want to shred the surroundings of with a ring of shrapnel, namely frag grenades and landmines.
+  *
+  * Note that grenades have extra handling for someone throwing themselves/being thrown on top of it, while landmines do not (obviously, it's a landmine!). See [/datum/component/pellet_cloud/proc/handle_martyrs()]
+  */
+/datum/component/pellet_cloud/proc/create_blast_pellets()
+	var/atom/A = parent
+	var/total_pellets_absorbed = 0
+
+	if(isgrenade(parent)) // handle_martyrs can reduce the radius and thus the number of pellets we produce if someone dives on top of a frag grenade
+		total_pellets_absorbed = handle_martyrs() // note that we can modify radius in this proc
+
+	if(radius < 1)
+		return
+
+	var/list/all_the_turfs_were_gonna_lacerate = RANGE_TURFS(radius, A) - RANGE_TURFS(radius-1, A)
+	num_pellets = all_the_turfs_were_gonna_lacerate.len + total_pellets_absorbed
+
+	for(var/T in all_the_turfs_were_gonna_lacerate)
+		var/turf/shootat_turf = T
+		pew(shootat_turf)
+
+/**
+  * handle_martyrs() is used for grenades that shoot shrapnel to check if anyone threw themselves/were thrown on top of the grenade, thus absorbing a good chunk of the shrapnel
+  *
+  * Between the time the grenade is armed and the actual detonation, we set var/list/bodies to the list of mobs currently on the new tile, as if the grenade landed on top of them, tracking if any of them move off the tile and removing them from the "under" list
+  * Once the grenade detonates, handle_martyrs() is called and gets all the new mobs on the tile, and add the ones not in var/list/bodies to var/list/martyrs
+  * We then iterate through the martyrs and reduce the shrapnel magnitude for each mob on top of it, shredding each of them with some of the shrapnel they helped absorb. This can snuff out all of the shrapnel if there's enough bodies
+  *
+  * Note we track anyone who's alive and client'd when they get shredded in var/list/purple_hearts, for achievement checking later
+  */
+/datum/component/pellet_cloud/proc/handle_martyrs()
+	var/list/martyrs = list()
+	for(var/mob/living/L in get_turf(parent))
+		if(!(L in bodies))
+			martyrs += L
+
+	var/magnitude_absorbed
+	var/total_pellets_absorbed
+
+	for(var/M in martyrs)
+		var/mob/living/L = M
```

Comment:
I never know if `L` means `living` or `limb`
Could you use something like `living_martyrs` or something like that?